src/dune_rules/merlin.ml

```diff
@@ -33,6 +33,7 @@ module Processed = struct
     ; src_dirs : Path.Set.t
     ; flags : string list
     ; extensions : string Ml_kind.Dict.t list
+    ; melc_flags : string list option
```

Comment:
What's the difference between `None` and `Some []`?

Reply:
`None` is used when we can't find the `melc_compiler`. I understand we don't want to fail at the merlin stage, like in other cases, e.g.

https://github.com/ocaml/dune/blob/437fabd9ae8254af1b62cdec23b6d091c90d6c94/src/dune_rules/merlin.ml#L345

Reply:
Yes, but then why not return `[]`?

Reply:
I guess we can, I thought `option` was more explicit. Updated in https://github.com/ocaml/dune/pull/6476/commits/dfff493f13a12a6802f826f4a9e56fde49385d5c.